core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/externaldatabase/ExternalDatabaseService.java

```diff
@@ -67,7 +68,15 @@ public void provisionDatabase(Cluster cluster, DatabaseAvailabilityType external
         AllocateDatabaseServerV4Request request = getDatabaseRequest(environment, externalDatabase, cluster);
         String databaseCrn;
         try {
-            databaseCrn = redbeamsClient.create(request).getResourceCrn();
+            Optional<DatabaseServerV4Response> existingDatabase = findExistingDatabase(cluster, environment.getCrn());
+            if (existingDatabase.isPresent()) {
+                String dbCrn = existingDatabase.get().getCrn();
+                LOGGER.debug("Found existing database with CRN {}", dbCrn);
+                databaseCrn = dbCrn;
+            } else {
+                LOGGER.debug("Requesting new database server creation");
+                databaseCrn = redbeamsClient.create(request).getResourceCrn();
```

Comment:
request creation should be moved right before this line

Reply:
ok